File: pyg-manager-web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){

    //修改指定商品状态
    this.updateAuditStatus=function(status,ids){
        return $http.get('../goods/updateAuditStatus/'+status+'/'+ids);
    };
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll');		
	};
	//分页 
	this.findPage=function(pageNum,pageSize){
		return $http.get('../goods/findPage?pageNum='+pageNum+'&pageSize='+pageSize);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../goods/delete?ids='+ids);
	};
	//搜索
	this.query=function(pageNum,pageSize,queryEntity){
		return $http.post('../goods/query?pageNum='+pageNum+"&pageSize="+pageSize, queryEntity);
	}    	
});

File: pyg-manager-interface/src/main/java/com/pyg/managerinterface/GoodsService.java
package com.pyg.managerinterface;
import java.util.List;

import com.pyg.bean.PageResult;
import com.pyg.groupEntity.Goods;
import com.pyg.pojo.TbGoods;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Goods goods);
	
	
	/**
	 * 修改
	 */
	public void update(TbGoods goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbGoods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void dele(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum, int pageSize);

	/**
	 * 修改指定商品状态
	 * @param status
	 * @param ids
	 */
	void updateAuditStatus(String status, Long[] ids);

	/**
	 *
	 * @param ids
	 * @param market
	 */
	void updateIsMarketable(Long[] ids, String market);
}
